graphql_compiler/tests/test_compiler.py

```diff
@@ -5282,3 +5282,42 @@ def test_complex_nested_optionals(self):
         '''
 
         check_test_data(self, test_data, expected_match, expected_gremlin)
+
+    def test_no_op_coercion(self):
+        test_data = test_input_data.no_op_coercion()
+
+        expected_match = '''
+        SELECT Animal__out_Animal_ParentOf__out_Animal_ParentOf___1.name
```

Comment:
nit: the other expected MATCH/Gremlin queries start one level indented from here, any reason not to do that here and below as well? Especially for Gremlin, you can break lines anywhere and all you need to do is simply follow what other testcases do as a convention.